libs/langchain/langchain/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,252 @@
+
+    async def _async_load(self) -> List[Document]:
+        """Load data from the loader using asyncio.
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+
+        event_loop = asyncio.get_event_loop()
+        tasks = [
+            event_loop.run_in_executor(
+                None,
+                self._partial_load,
+                loader_args,
+            )
+            for loader_args in self.loaders_args
+        ]
+        if self.show_progress:
+            try:
+                from tqdm.asyncio import tqdm
+            except ImportError as err:
+                raise ImportError(
+                    "You must install tqdm to use show_progress=True."
+                    "You can install tqdm with `pip install tqdm`."
+                ) from err
+
+            results = await tqdm.gather(*tasks)
+        else:
+            results = await asyncio.gather(*tasks)
+        return [doc for docs in results for doc in docs]
+
+    def _load_thread(self, max_workers: int) -> List[Document]:
+        """Load data from the loader using threads executor.
+
+        Args:
+            max_workers (int): number of workers
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+
+        with ThreadPoolExecutor(max_workers) as executor:
+            docs_list = make_iterator_list(
+                iterable=executor.map(
+                    self._partial_load,
+                    self.loaders_args,
+                ),
+                args_length=len(self.loaders_args),
+                show_progress=self.show_progress,
+            )
+
+        return [doc for docs in docs_list for doc in docs]
+
+    def _load_process(self, max_workers: int) -> List[Document]:
+        """Load data from the loader using process pool executor.
+
+        Args:
+            max_workers (int): number of workers
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+
+        with multiprocessing.Pool(max_workers) as pool:
+            docs_list = make_iterator_list(
+                iterable=pool.imap(
+                    self._partial_load,
+                    self.loaders_args,
+                ),
+                args_length=len(self.loaders_args),
+                show_progress=self.show_progress,
+            )
+
+        return [doc for docs in docs_list for doc in docs]
+
+    def _load_sequential(self) -> List[Document]:
+        """Load data from the loader sequentially.
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+        docs_list = make_iterator_list(
+            iterable=map(self._partial_load, self.loaders_args),
+            args_length=len(self.loaders_args),
+            show_progress=self.show_progress,
+        )
+
+        return [doc for docs in docs_list for doc in docs]
+
+    @staticmethod
+    def _load_loader(
```

Comment:
nit: it's confusing when it's part of the namespace since it's invoked from the init, and it's not obvious from the init that it's a staticmethod rather than a method. Could we move as a private function in the module namespace instead of the classnamespace?

Reply:
You're right
Done in last commit